pages/api/moralis/pinFile.ts

```diff
@@ -9,13 +10,34 @@ export const config = {
   },
 };
 
+const toBase64 = (file: ArrayBuffer) => {
+    return Buffer.from(file).toString('base64');
+  };
+
 export default async function handler(
   request: NextApiRequest,
   response: NextApiResponse
 ) {
-  const { b64File, name } = request.body;
+  const { url, name } = request.body;
+
+  const fileResponse = await fetch(url);
+  const fileResult = await fileResponse.arrayBuffer();
+
+  let b64File = toBase64(fileResult);
```

Comment:
You actually don't have to convert it to base64 anymore now that you have the actual file. That was done before so that we could put it in the request body. 

https://docs.moralis.io/moralis-dapp/files/ipfs#saving-files

Make `pinFile` look more like this. You can just pass the file to it. 

```
const data = fileInput.files[0];
const file = new Moralis.File(data.name, data);
await file.saveIPFS();
```

Reply:
I am getting an ArrayBuffer tho, it doesn't seem like Moralis supports that.

Reply:
You're getting a 'Blob', which should be possible to convert to a file 

https://stackoverflow.com/questions/27159179/how-to-convert-blob-to-file-in-javascript